packages/service-library/src/servicelib/archiving_utils.py

```diff
@@ -118,32 +135,59 @@ async def unarchive_dir(
 
 def _serial_add_to_archive(
     dir_to_compress: Path, destination: Path, compress: bool, store_relative_path: bool
-) -> bool:
-    compression = zipfile.ZIP_DEFLATED if compress else zipfile.ZIP_STORED
-    with zipfile.ZipFile(destination, "w", compression=compression) as zip_file_handler:
-        files_to_compress_generator = _full_file_path_from_dir_and_subdirs(
-            dir_to_compress
-        )
-        for file_to_add in files_to_compress_generator:
-            try:
-                file_name_in_archive = (
-                    _strip_directory_from_path(file_to_add, dir_to_compress)
-                    if store_relative_path
-                    else file_to_add
-                )
-                zip_file_handler.write(file_to_add, file_name_in_archive)
-            except ValueError:
-                log.exception("Could write files to archive, please check logs")
-                return False
-    return True
+) -> Union[None, Exception]:
+    try:
+        compression = zipfile.ZIP_DEFLATED if compress else zipfile.ZIP_STORED
+        with zipfile.ZipFile(
+            destination, "w", compression=compression
+        ) as zip_file_handler:
+            files_to_compress_generator = _full_file_path_from_dir_and_subdirs(
+                dir_to_compress
+            )
+            for file_to_add in files_to_compress_generator:
+                try:
+                    file_name_in_archive = (
+                        _strip_directory_from_path(file_to_add, dir_to_compress)
+                        if store_relative_path
+                        else file_to_add
+                    )
+
+                    # because surrogates are not allowed in zip files,
+                    # replacing them will ensure errors will not happen.
+                    escaped_file_to_add = _surrogate_replace_in_path(file_to_add)
+                    escaped_file_name_in_archive = _surrogate_replace_in_path(
+                        file_name_in_archive
+                    )
+
+                    # ensure original file is not changed by
+                    # creteing a symlink with an archivable filename by zipfile
```

Comment:
i do not fully understand this: how does creating a symlink prevents the original file from changing?
And the simlink is not added to the zip, right?

Reply:
Let me figure out what I was truing to do here. Because I might have gotten myself confused by some other issue.

Reply:
This entire symlink is not required. As the issue was with the name of the file in the archive and not the name of the source file